src/librustc_errors/diagnostic_builder.rs

```diff
@@ -167,26 +170,51 @@ impl<'a> DiagnosticBuilder<'a> {
                                                   sp: S,
                                                   msg: &str)
                                                   -> &mut Self);
+
+    #[deprecated(note = "Use `span_suggestion_short_with_applicability`")]
     forward!(pub fn span_suggestion_short(&mut self,
                                           sp: Span,
                                           msg: &str,
                                           suggestion: String)
                                           -> &mut Self);
+
+    #[deprecated(note = "Use `multipart_suggestion_with_applicability`")]
     forward!(pub fn multipart_suggestion(
         &mut self,
         msg: &str,
         suggestion: Vec<(Span, String)>
     ) -> &mut Self);
+
+    #[deprecated(note = "Use `span_suggestion_with_applicability`")]
     forward!(pub fn span_suggestion(&mut self,
                                     sp: Span,
                                     msg: &str,
                                     suggestion: String)
                                     -> &mut Self);
+
+    #[deprecated(note = "Use `span_suggestions_with_applicability`")]
     forward!(pub fn span_suggestions(&mut self,
                                      sp: Span,
                                      msg: &str,
                                      suggestions: Vec<String>)
                                      -> &mut Self);
+
+    pub fn multipart_suggestion_with_applicability(&mut self,
+                                              msg: &str,
+                                              suggestion: Vec<(Span, String)>,
+                                              applicability: Applicability)
+                                              -> &mut Self {
```

Comment:
Indentation:

```
pub fn foo(
    &mut self,
    bar: &str,
) -> &mut Self {
```

Reply:
Changed a bit more globally, the macro included.